pkgs/development/compilers/graalvm/community-edition.nix

```diff
@@ -60,40 +72,57 @@ let
 
            mkdir -p $out
            arr=($srcs)
-           tar xf ''${arr[0]} -C $out --strip-components=1
+
+           # The tarball on Linux has the following directory structure:
+           #
+           #   graalvm-ce-java11-20.3.0/*
+           #
+           # while on Darwin it looks like this:
+           #
+           #   graalvm-ce-java11-20.3.0/Contents/Home/*
+           #
+           # We therefor use --strip-components=1 vs 3 depending on the platform.
```

Comment:
hmm... maybe a quick sanity-check to test that this is the exact directory structure we have before running the `tar` command so if a future upgrade breaks this assumption we get a failing build rather than a broken directory structure?

Reply:
The directory structure is *inside* the graalvm tarball, if that changes in the future then the `unpackPhase` will have to be updated.

Trying to introduce some smart logic here that will behave correctly in the future will be brittle and more confusing than helpful, I think.

Reply:
yeah, my worry is that the unpackPhase would *work*, but produce incorrect results without the build failing. I was thinking about adding some `test -d` checks to make sure the directory structure is what we expect it to.

If my worries about the build not failing are unfounded, feel free to disregard.

Reply:
Added a sanity check, thanks!

Reply:
looks really good, thanks!